runtime-modules/project-token/src/tests/bonding_curve.rs

```diff
@@ -0,0 +1,1079 @@
+            );
+        })
+}
+
+#[test]
+fn unbonding_ok_with_crt_issuance_decreased() {
+    let token_id = token!(1);
+    let (user_account_id, user_balance) = (member!(2).1, joy!(5_000_000));
+    let (creator_id, _) = member!(1);
+    build_default_test_externalities_with_balances(vec![(user_account_id, user_balance)])
+        .execute_with(|| {
+            IssueTokenFixture::default()
+                .with_creator_id(creator_id)
+                .execute_call()
+                .unwrap();
+            ActivateAmmFixture::default().execute_call().unwrap();
+            BondFixture::default().execute_call().unwrap();
+            let supply_pre = Token::token_info_by_id(token_id).total_supply;
+
+            UnbondFixture::default()
+                .with_amount(DEFAULT_UNBONDING_AMOUNT)
+                .execute_call()
+                .unwrap();
+
+            let supply_post = Token::token_info_by_id(token_id).total_supply;
+            assert_eq!(supply_pre - supply_post, DEFAULT_UNBONDING_AMOUNT);
+        })
+}
+
+#[test]
+fn unbonding_fails_with_amm_treasury_not_having_sufficient_usable_joy_required() {
+    let token_id = token!(1);
+    let (user_account_id, user_balance) = (member!(2).1, joy!(5_000_000));
+    build_default_test_externalities_with_balances(vec![(user_account_id, user_balance)])
+        .execute_with(|| {
+            IssueTokenFixture::default().execute_call().unwrap();
+            ActivateAmmFixture::default().execute_call().unwrap();
+            BondFixture::default().execute_call().unwrap();
+            // setting the balance of teh bonding curve reserve to 0
+            Balances::set_balance(
+                Origin::root(),
+                Token::module_bonding_curve_reserve_account(token_id),
+                Balance::zero(),
+                Balance::zero(),
+            )
+            .unwrap();
+
+            let result = UnbondFixture::default().execute_call();
+
+            assert_err!(result, Error::<Test>::InsufficientJoyBalance);
+        })
+}
+
+#[test]
+fn unbonding_ok_with_user_joy_balance_correctly_increased() {
+    let (_, user_account) = member!(2);
+    let (user_account_id, user_balance) = (member!(2).1, joy!(5_000_000));
+    build_default_test_externalities_with_balances(vec![(user_account_id, user_balance)])
+        .execute_with(|| {
+            IssueTokenFixture::default()
+                .with_empty_allocation()
+                .execute_call()
+                .unwrap();
+            ActivateAmmFixture::default()
+                .with_creator_reward(Permill::zero())
+                .execute_call()
+                .unwrap();
+            BondFixture::default().execute_call().unwrap();
+            let user_reserve_pre = Balances::usable_balance(user_account);
+
+            UnbondFixture::default().execute_call().unwrap();
+
+            let user_reserve_post = Balances::usable_balance(user_account);
+            let correctly_computed_joy_amount = 95;
```

Comment:
Please add an explainer (comment) as to where this value comes from

Reply:
I added an helper function to make the computation clearer